### PR TITLE
[PROF-9132] document Go custom profiler annotations

### PR DESCRIPTION
**Pull request by nsrip-dd**
### What does this PR do? What is the motivation?

We've added a public API to support user-provided annotations which
whill appear in Go CPU and goroutine profiles. Document how to use them.
The advice on the "Isolate Outliers in Monolothic Services" is relevant
to this, and most of it applies to Go just as much as Java. Put the
Go-specific bits there.

### Merge instructions

- [ ] Please merge after reviewing

### CODE REVIEW

**Review by jhgilbert — APPROVED** [review]

**Comment by gturbat** [comment]
I think we can group Java and Go under the same header by using

```
{{< programming-lang-wrapper langs="java,go" >}}
{{< programming-lang lang="java" >}}
...
{{< /programming-lang >}}
{{< programming-lang lang="go" >}}
...
{{< /programming-lang-wrapper >}}
```

This way it will be easier to add other languages.

**Comment by nsrip-dd** [comment]
> I think we can group Java and Go under the same header by using the tab pattern:

Good idea, thanks Guillaume! Done.

@jhgilbert thanks for the review! Would you mind taking a look to make sure I got it right? The text is the same, and it looks fine to me in a preview.

File: content/en/profiler/guide/isolate-outliers-in-monolithic-services.md [hunk 1]
```diff
@@ -74,20 +74,50 @@ The APM `Trace operation` attribute lets you filter and group a flame graph with
...
 In the previous image, notice that the `ModelTraining` operation is taking more CPU time than its primary use in the `GET /train` endpoint, so it must be used elsewhere. Click the operation name to determine where else it is used. In this case, `ModelTraining` is also use by `POST /update_model`.
 
-## Isolate your own business logic (Java)
+
+## Isolate your own business logic
+
+{{< programming-lang-wrapper langs="java,go" >}}
+{{< programming-lang lang="java">}}
 
 Endpoint and operation isolation is available in your profiles by default, but you may want to isolate a different piece of logic. For example, if the monolith is sensitive to specific customers, you can add a custom filter to the profiles:
```
Comment (gturbat) [inline content/en/profiler/guide/isolate-outliers-in-monolithic-services.md:83]: I think we can keep this introduction above the tabs. It is relevant to both languages

File: content/en/profiler/guide/isolate-outliers-in-monolithic-services.md [hunk 2]
```diff
@@ -74,20 +74,50 @@ The APM `Trace operation` attribute lets you filter and group a flame graph with
...
 
-{{< code-block lang="java" >}}
+```java
 try (var scope = Profiling.get().newScope()) {
    scope.setContextValue("customer_name", <the customer name value>);
    <logic goes here>
 }
-{{< /code-block >}}
+```
+
```
Comment (gturbat) [inline content/en/profiler/guide/isolate-outliers-in-monolithic-services.md:91]:
```suggestion
Then, open CPU, Exceptions or Wall Time profiles for your service and select the customer_name value you're interested in under the CPU time by dropdown.
```

File: content/en/profiler/guide/isolate-outliers-in-monolithic-services.md [hunk 3]
```diff
@@ -74,20 +74,50 @@ The APM `Trace operation` attribute lets you filter and group a flame graph with
...
+## Isolate your own business logic
+
+{{< programming-lang-wrapper langs="java,go" >}}
+{{< programming-lang lang="java">}}
 
 Endpoint and operation isolation is available in your profiles by default, but you may want to isolate a different piece of logic. For example, if the monolith is sensitive to specific customers, you can add a custom filter to the profiles:
 
-{{< code-block lang="java" >}}
+```java
```
Comment (gturbat) [inline content/en/profiler/guide/isolate-outliers-in-monolithic-services.md:85]:
```suggestion
To add annotations, use profiler labels like so:
```java
```

**Review by gturbat — APPROVED** [review]

**Comment by nsrip-dd** [comment]
@jhgilbert would you please take another look, and merge if it's good to go?

**Pull request merged**